src/components/loader-spinner/loader-spinner.component.tsx

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+import tagComponent, {
+  TagProps,
+} from "../../__internal__/utils/helpers/tags/tags";
+import {
+  StyledSpinnerWrapper,
+  StyledLabel,
+  StyledSpinnerCircleSvg,
+} from "./loader-spinner.style";
+import useLocale from "../../hooks/__internal__/useLocale";
+import useMediaQuery from "../../hooks/useMediaQuery";
+import Typography from "../typography";
+
+export type LoaderSpinnerSizes = "XS" | "S" | "M" | "L" | "XL";
+
+export type LoaderSpinnerColors =
+  | "action"
+  | "neutral"
+  | "inverse"
+  | "gradient-grey"
+  | "gradient-white";
+
+export interface LoaderSpinnerProps extends MarginProps, TagProps {
+  /**
+   * The size prop allows a specific size to be set, ranging from
+   * `XS` to `XL`
+   */
+  size?: LoaderSpinnerSizes;
+  /**
+   * If set to `false` no visual label will be displayed, however
+   * a visually hidden label will still be available for assistive technologies
+   */
+  showSpinnerLabel?: boolean;
+  /**
+   * The color prop can be used to change the color of the component.
+   * Typically both the outer and inner spinner will change color,
+   * however there will still be sufficient contrast between them
+   */
+  color?: LoaderSpinnerColors;
+  /**
+   * The `labelColor` prop can be used to change the font color of the
+   * rendered label
+   */
+  labelColor?: "dark" | "light";
+  /** If set to `false` all motion will be suspended */
+  motion?: boolean;
```

Comment:
comment (non-blocking): we could name this a bit more intuitively for example `hasMotion` etc